### PR TITLE
refactor(a2-1099): create question prop type d

### PR DESCRIPTION
**Pull request by Si3rr4wow**
## Ticket Number

https://pins-ds.atlassian.net/browse/AAPD-

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- [ ] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/Planning-Inspectorate/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/forms-web-app/src/dynamic-forms/middleware/redirect-to-unanswered-question.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /**
- * @param {[((question: import('../section').Question, answer: unknown) => boolean)]} conditions
+ * @param {[((question: import('../section').Question, journeyResponse: import('../journey-response').JourneyResponse) => boolean)]} conditions
```
Comment (Si3rr4wow) [inline packages/forms-web-app/src/dynamic-forms/middleware/redirect-to-unanswered-question.js:2]: just aligning this with it's actual usage

File: packages/forms-web-app/src/dynamic-forms/dynamic-components/list-add-more/question.js [hunk 2]
```diff
@@ -64,11 +75,7 @@ class ListAddMoreQuestion extends Question {
 			validators
 		});
 
-		if (!subQuestion || !(subQuestion instanceof AddMoreQuestion)) {
-			throw new Error('addMore is mandatory');
-		}
```
Comment (Si3rr4wow) [inline packages/forms-web-app/src/dynamic-forms/dynamic-components/list-add-more/question.js:69]: This gets caught by the type system. I'm hoping to get questions.js to zero type issues soon

File: packages/forms-web-app/src/dynamic-forms/dynamic-components/boolean/question.js [hunk 3]
```diff
@@ -25,7 +25,7 @@ class BooleanQuestion extends RadioQuestion {
 		description,
 		html,
 		validators,
-		type = 'radio',
+		interfaceType = 'radio',
```
Comment (Si3rr4wow) [inline packages/forms-web-app/src/dynamic-forms/dynamic-components/boolean/question.js:28]: This frees up the key `type` to describe the type of question. See question-props.d.ts

**Review by beejjacobs — COMMENTED** [review]
Change looks good to me - the type definitions are clear, too.

**Review by bryn-pins — APPROVED** [review]

**Review by bryn-pins — APPROVED** [review]

**Pull request merged**